crsqlite_extension/src/storage.c

```diff
@@ -0,0 +1,106 @@
+#include <stdio.h>
+#include <string.h>
+#include <stdarg.h>
+
+#include <sqlite3ext.h>
+
+#ifndef sqlite3_api
+SQLITE_EXTENSION_INIT3
+#endif
+
+#define COLUMN_NAME 1
+#define COLUMN_TYPE 2
+#define COLUMN_NOT_NULL 3
+#define COLUMN_DFLT_VAL 4
+#define COLUMN_PK 5
+
+ /*
+** Construct one or more SQL statements from the format string given
+** and then evaluate those statements. The success code is written
+** into *pRc.
+**
+** If *pRc is initially non-zero then this routine is a no-op.
+*/
+static void storageDbExec(
+  int *pRc,              /* Success code */
+  sqlite3 *db,           /* Database in which to run SQL */
+  const char *zFormat,   /* Format string for SQL */
+  ...                    /* Arguments to the format string */
+){
+  va_list ap;
+  char *zSql;
+  if( *pRc ) return;
+  va_start(ap, zFormat);
+  zSql = sqlite3_vmprintf(zFormat, ap);
+  //printf("Executing SQL statement: %s\n", zSql);
+  va_end(ap);
+  if( zSql==0 ){
+    *pRc = SQLITE_NOMEM;
+    
+  }else{
+    *pRc = sqlite3_exec(db, zSql, 0, 0, 0);
+    sqlite3_free(zSql);
+  }
+}
+
+
+/* Initites the real database table(s) that backs the virtual table*/
+int init_storage(
+  sqlite3 *db,
+  const char* ttbl,
+  const char* createTableArgs,
+  char **pzErr
+)
+{
+  
+  //Create the table with "normal" schema
```

Comment:
comment is a bit misleading.

You're creating the `cr` table but without the version columns. Given its the `cr` table, I'm not sure I'd call it the "normal" table.